models/src/main/scala/coop/rchain/models/SerializeInstances.scala

```diff
@@ -0,0 +1,30 @@
+package coop.rchain.models
+
+import cats.syntax.either._
+import com.trueaccord.scalapb.{GeneratedMessage, GeneratedMessageCompanion}
+
+trait SerializeInstances {
+  case class rhoInstanceWrapper[T <: GeneratedMessage with com.trueaccord.scalapb.Message[T]](
+      companion: GeneratedMessageCompanion[T])
+      extends Serialize[T] {
+    override def encode(a: T): Array[Byte] = companion.toByteArray(a)
+
+    override def decode(bytes: Array[Byte]): Either[SerializeError, T] =
+      Either
+        .catchNonFatal(companion.parseFrom(bytes))
+        .leftMap(SerializeError.apply)
+  }
```

Comment:
After changing `Serialize`, this becomes:

```scala
  case class rhoInstanceWrapper[T <: GeneratedMessage with com.trueaccord.scalapb.Message[T]](
      companion: GeneratedMessageCompanion[T])
      extends Serialize[T] {
    override def encode(a: T): Array[Byte] = companion.toByteArray(a)

    override def decode(bytes: Array[Byte]): Either[Throwable, T] =
      Either.catchNonFatal(companion.parseFrom(bytes))
  }
```

Reply:
Done.